### ISSUE TITLE
Missing django.forms.util module in pypi package (1.0.3)

### ISSUE THREAD

**Issue by Germano0** [bradjasper/django-jsonfield#169]
Hi, I am packaging python-django-jsonfields to include it into Fedora repositories. While making the RPM package I obtained the error message `ImportError: No module named util`.
Here the complete build log

```
[user@machine rpmbuild]$ rpmbuild -ba SPECS/python-django-jsonfield.spec 
Esecuzione(%prep) in corso: /bin/sh -e /var/tmp/rpm-tmp.vrYGR6
+ umask 022
+ cd /home/user/rpmbuild/BUILD
+ cd /home/user/rpmbuild/BUILD
+ rm -rf jsonfield-1.0.3
+ /usr/bin/gzip -dc /home/user/rpmbuild/SOURCES/jsonfield-1.0.3.tar.gz
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd jsonfield-1.0.3
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ exit 0
Esecuzione(%build) in corso: /bin/sh -e /var/tmp/rpm-tmp.9kOBbT
+ umask 022
+ cd /home/user/rpmbuild/BUILD
+ cd jsonfield-1.0.3
+ CFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic'
+ /usr/bin/python2 setup.py build '--executable=/usr/bin/python2 -s'
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'include_package_data'
  warnings.warn(msg)
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'tests_require'
  warnings.warn(msg)
running build
running build_py
creating build
creating build/lib
creating build/lib/jsonfield
copying jsonfield/encoder.py -> build/lib/jsonfield
copying jsonfield/models.py -> build/lib/jsonfield
copying jsonfield/__init__.py -> build/lib/jsonfield
copying jsonfield/subclassing.py -> build/lib/jsonfield
copying jsonfield/tests.py -> build/lib/jsonfield
copying jsonfield/fields.py -> build/lib/jsonfield
+ CFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic'
+ /usr/bin/python3 setup.py build '--executable=/usr/bin/python3 -s'
/usr/lib64/python3.5/distutils/dist.py:261: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
/usr/lib64/python3.5/distutils/dist.py:261: UserWarning: Unknown distribution option: 'tests_require'
  warnings.warn(msg)
/usr/lib64/python3.5/distutils/dist.py:261: UserWarning: Unknown distribution option: 'include_package_data'
  warnings.warn(msg)
running build
running build_py
+ exit 0
Esecuzione(%install) in corso: /bin/sh -e /var/tmp/rpm-tmp.dx2NWF
+ umask 022
+ cd /home/user/rpmbuild/BUILD
+ '[' /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64 '!=' / ']'
+ rm -rf /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64
++ dirname /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64
+ mkdir -p /home/user/rpmbuild/BUILDROOT
+ mkdir /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64
+ cd jsonfield-1.0.3
+ CFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic'
+ /usr/bin/python2 setup.py install -O1 --skip-build --root /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'include_package_data'
  warnings.warn(msg)
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'tests_require'
  warnings.warn(msg)
running install
running install_lib
creating /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr
creating /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib
creating /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7
creating /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages
creating /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield
copying build/lib/jsonfield/encoder.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield
copying build/lib/jsonfield/models.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield
copying build/lib/jsonfield/__init__.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield
copying build/lib/jsonfield/subclassing.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield
copying build/lib/jsonfield/tests.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield
copying build/lib/jsonfield/fields.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield/encoder.py to encoder.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield/models.py to models.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield/__init__.py to __init__.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield/subclassing.py to subclassing.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield/tests.py to tests.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield/fields.py to fields.pyc
writing byte-compilation script '/tmp/tmp1zQZWo.py'
/usr/bin/python2 -O /tmp/tmp1zQZWo.py
removing /tmp/tmp1zQZWo.py
running install_egg_info
Writing /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7/site-packages/jsonfield-1.0.3-py2.7.egg-info
+ CFLAGS='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic'
+ /usr/bin/python3 setup.py install -O1 --skip-build --root /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64
/usr/lib64/python3.5/distutils/dist.py:261: UserWarning: Unknown distribution option: 'include_package_data'
  warnings.warn(msg)
/usr/lib64/python3.5/distutils/dist.py:261: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
/usr/lib64/python3.5/distutils/dist.py:261: UserWarning: Unknown distribution option: 'tests_require'
  warnings.warn(msg)
running install
running install_lib
creating /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5
creating /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages
creating /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield
copying build/lib/jsonfield/encoder.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield
copying build/lib/jsonfield/models.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield
copying build/lib/jsonfield/__init__.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield
copying build/lib/jsonfield/subclassing.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield
copying build/lib/jsonfield/tests.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield
copying build/lib/jsonfield/fields.py -> /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield/encoder.py to encoder.cpython-35.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield/models.py to models.cpython-35.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield/__init__.py to __init__.cpython-35.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield/subclassing.py to subclassing.cpython-35.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield/tests.py to tests.cpython-35.pyc
byte-compiling /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield/fields.py to fields.cpython-35.pyc
writing byte-compilation script '/tmp/tmp3w43906z.py'
/usr/bin/python3 -O /tmp/tmp3w43906z.py
removing /tmp/tmp3w43906z.py
running install_egg_info
Writing /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5/site-packages/jsonfield-1.0.3-py3.5.egg-info
+ /usr/lib/rpm/find-debuginfo.sh --strict-build-id -m --run-dwz --dwz-low-mem-die-limit 10000000 --dwz-max-die-limit 110000000 /home/user/rpmbuild/BUILD/jsonfield-1.0.3
/usr/lib/rpm/sepdebugcrcfix: Updated 0 CRC32s, 0 CRC32s did match.
find: 'debug': No such file or directory
+ '[' noarch = noarch ']'
+ case "${QA_CHECK_RPATHS:-}" in
+ /usr/lib/rpm/check-buildroot
+ /usr/lib/rpm/brp-compress
+ /usr/lib/rpm/brp-strip-static-archive /usr/bin/strip
+ /usr/lib/rpm/brp-python-bytecompile /usr/bin/python 1
Bytecompiling .py files below /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python2.7 using /usr/bin/python2.7
Bytecompiling .py files below /home/user/rpmbuild/BUILDROOT/python-django-jsonfield-1.0.3-1.fc24.x86_64/usr/lib/python3.5 using /usr/bin/python3.5
+ /usr/lib/rpm/brp-python-hardlink
+ /usr/lib/rpm/redhat/brp-java-repack-jars
Esecuzione(%check) in corso: /bin/sh -e /var/tmp/rpm-tmp.UT8CBv
+ umask 022
+ cd /home/user/rpmbuild/BUILD
+ cd jsonfield-1.0.3
+ /usr/bin/python2 setup.py test
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'include_package_data'
  warnings.warn(msg)
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'install_requires'
  warnings.warn(msg)
/usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'tests_require'
  warnings.warn(msg)
running test
Creating test database for alias 'default'...
E
======================================================================
ERROR: jsonfield.tests (unittest.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: jsonfield.tests
Traceback (most recent call last):
  File "/usr/lib64/python2.7/unittest/loader.py", line 254, in _find_tests
    module = self._get_module_from_name(name)
  File "/usr/lib64/python2.7/unittest/loader.py", line 232, in _get_module_from_name
    __import__(name)
  File "/home/user/rpmbuild/BUILD/jsonfield-1.0.3/jsonfield/tests.py", line 12, in <module>
    from django.forms.util import ValidationError
ImportError: No module named util


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
Destroying test database for alias 'default'...
errore: Stato d'uscita errato da /var/tmp/rpm-tmp.UT8CBv (%check)


Errori di compilazione RPM:
    Stato d'uscita errato da /var/tmp/rpm-tmp.UT8CBv (%check)
```

**Comment by yourcelf** [comment]
It looks as thought the tests have not been updated for Django v1.9+.  Django renamed all `util` modules to `utils` for consistency: https://docs.djangoproject.com/en/dev/internals/deprecation/#deprecation-removed-in-1-9.  A fix to the test file which would be cross compatible across ancient and recent Django versions:

```
try:
    from django.forms.utils import ValidationError
except ImportError:
    from django.forms.util import ValidationError
```

**Comment by Germano0** [comment]
https://github.com/bradjasper/django-jsonfield/commit/d2d2c58815501b13d8bc7988c06594135afae39f
fixed it :-)

**Comment by dmkoch** [comment]
This should be fixed for the next release following 1.0.3
